core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMaterializedViewDefinition.java

```diff
@@ -142,4 +163,16 @@ public String toString()
             return name + " " + type;
         }
     }
+
+    public enum RefreshMode
+    {
+        /**
+         * Refresh materialized view using {@link ConnectorMetadata#refreshMaterializedView} method call
+         */
+        DELEGATE_TO_CONNECTOR,
+        /**
+         * Refresh materialized view by performing insert query to storage table
+         */
+        INSERT_QUERY,
```

Comment:
Why an enum? The MV is either refreshed by the connector or by the engine. Also, INSERT_QUERY many not be appropriate once we start supporting more granular refreshes that require DELETE/UPDATE.

Reply:
> Why an enum? The MV is either refreshed by the connector or by the engine.

What would be alternative to enum? Enum means exactly this: `The MV is either refreshed by the connector or by the engine.`. Do you mean enum constant names should be different?

Reply:
An alternative would be a simple boolean flag, but it depends on whether it's reasonable to expect additional modes. I can't think of any reason for this, though.

Reply:
should I change it to boolean though?

Reply:
changed to boolean